File: design-pattern-go/src/patterns/structural/adapter/AdapterIntToString.go
package adapter

import (
	"strconv"
)

type AdapterIntToString struct {
	service ServicePrintString
}

func (ad *AdapterIntToString) AddService(
	service ServicePrintString) *AdapterIntToString {
	ad.service = service
	return ad
}
func (ad *AdapterIntToString) Apply(input int) string {
	return ad.service.Apply(strconv.Itoa(input))
}

File: design-pattern-java/app/src/main/java/design/pattern/java/patterns/structural/adapter/AdapterIntToString.java
package design.pattern.java.patterns.structural.adapter;

import java.util.function.Function;


public class AdapterIntToString implements Function<Integer, String>{

    private final Function<String, String> service;

    public AdapterIntToString(Function<String, String> service) {
        this.service = service;
    }

    @Override
    public String apply(Integer input) {
        return this.service.apply(String.valueOf(input));
    }
    
}
